frontend/src/Search.js

```diff
@@ -242,65 +296,13 @@ class Search extends Component {
           if(search_query === this.state.search || search_query === '$') {
             this.setState({data: data});
             // Render search results
-            ReactDOM.render(
-              data.map((course)=> {
-                let termCode = convertSemToTermCode(course["semester_list"][course["semester_list"].length - 1]);
-                return <li key={course["id"]} id={course["id"]} className={"course-display " + course["semester"]} dist_area={course["dist_area"]}>
-                <p className="course-name">{course["listing"]}</p>
-                <i className="fas fa-times-circle delete-course"></i>
-                <a href={"https://registrar.princeton.edu/course-offerings/course_details.xml?courseid=" + course["id"] + "&term=" + termCode} target="_blank"><i className="fas fa-info-circle fa-lg fa-fw course-info"></i></a>
-                <a href={"https://reg-captiva.princeton.edu/chart/index.php?terminfo=" + termCode + "&courseinfo=" + course["id"]} target="_blank"><i className="fas fa-chart-bar fa-lg fa-fw course-eval"></i></a>
-                <p className="course-title">{course["title"]}</p>
-                <p className="course-semester">{"Offered in " + convertSemListToReadableForm(course["semester_list"])}</p>
-                </li>
-              }),
-            document.getElementById('display-courses')
-            )
-            ReactDOM.render(<span id='search-count-num'>{data.length}</span>, document.getElementById('search-count'))
+            ReactDOM.render(returnSearchList(data), document.getElementById('display-courses'));
+            ReactDOM.render(<span id='search-count-num'>{data.length}</span>, document.getElementById('search-count'));
             $('#spinner').css('display', 'none');
           }
         }.bind(this)
     });
 
-    /* Converts semester to term code */
```

Comment:
Moved functions outside so returnSearchList will have access to it. I also moved returnSearchList outside because requirements.js needed access to that function.